### ISSUE TITLE
Windows: Crash when running process.exit()

### ISSUE THREAD

**Issue by bpasero** [atom/electron#3580]
In VS Code when I call process.exit() from the main side I get the following trace. I have not been able to reproduce on a bare Electron 0.34.1 though. Any ideas what I could look for to nail this down?

C:\GitDevelopment\monaco>       (No symbol) [0x0046D436]
        (No symbol) [0x0036B0FF]
        (No symbol) [0x0036AC8A]
        uv_async_send [0x6403A4FD+1003]
        _get_flsindex [0x63A3C01D+111]
        _get_flsindex [0x63A3C001+83]
        BaseThreadInitThunk [0x756E7C04+36]
        RtlInitializeExceptionChain [0x7706AD1F+143]
        RtlInitializeExceptionChain [0x7706ACEA+90]

**Comment by zcbenz** [comment]
I think we have race conditions when there are lots of Node tasks running while closing the app, here is a similar crash report I saw before:

```
Received signal 11 SEGV_MAPERR 000000000208
#0 0x7fb45e46aca8 <unknown>
#1 0x7fb456e83740 <unknown>
#2 0x7fb468b489c3 uv__async_send
#3 0x7fb468b48954 uv_async_send
#4 0x0000006da78c atom::NodeBindings::WakeupEmbedThread()
#5 0x0000006da6f8 atom::NodeBindings::~NodeBindings()
#6 0x0000006dd505 atom::NodeBindingsLinux::~NodeBindingsLinux()
#7 0x0000006dd529 atom::NodeBindingsLinux::~NodeBindingsLinux()
#8 0x000000693012 base::DefaultDeleter<>::operator()()
#9 0x000000692fc0 base::internal::scoped_ptr_impl<>::~scoped_ptr_impl()
#10 0x000000692145 scoped_ptr<>::~scoped_ptr()
#11 0x0000006e82a8 atom::AtomRendererClient::~AtomRendererClient()
#12 0x0000006e8319 atom::AtomRendererClient::~AtomRendererClient()
#13 0x00000068de32 base::DefaultDeleter<>::operator()()
#14 0x00000068e310 base::internal::scoped_ptr_impl<>::~scoped_ptr_impl()
#15 0x00000068d8c5 scoped_ptr<>::~scoped_ptr()
#16 0x00000068d13a atom::AtomMainDelegate::~AtomMainDelegate()
#17 0x00000068bfa7 main
```

The crashing line is:
https://github.com/atom/electron/blob/v0.35.1/atom/common/node_bindings.cc#L122

**Comment by zcbenz** [comment]
I think this had been fixed by #3356 in v0.34.3, and #3570 is not related to this.
